pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -28,7 +44,7 @@ func Validate(ic *types.InstallConfig) error {
 // ValidateForProvisioning performs platform validation specifically for installer-
 // provisioned infrastructure. In this case, self-hosted networking is a requirement
 // when the installer creates infrastructure for vSphere clusters.
-func ValidateForProvisioning(ic *types.InstallConfig) error {
+func ValidateForProvisioning(ic *types.InstallConfig, certificates ...x509.Certificate) error {
```

Comment:
it looks like this new `certificates` parameter is not being used, or maybe I'm just missing it

Reply:
@patrickdillon for the new validation methods to pass unit testing we need a vSphere API - this is provided by the simulator. The next problem is the simulator when started with SSL is a invalid self-signed certificate. The installer requires this certificate to be valid. The way I worked around this was to create a CA and certificate, pass the CA to `CreateVSphereClients`. (maybe there is a better way, all ears ;-) )